### PR TITLE
Android initChannel and sendMessage method added

### PR DESCRIPTION
**Pull request by ronit-kadwane**
GoogleCast.initChannel() and Google.sendMessage() methods added for android.

### CODE REVIEW

File: android/src/main/java/com/reactnative/googlecast/GoogleCastModule.java [hunk 1]
```diff
@@ -14,288 +16,352 @@
...
+    protected static final String MEDIA_STATUS_UPDATED =
+            "GoogleCast:MediaStatusUpdated";
+    protected static final String MEDIA_PLAYBACK_STARTED =
+            "GoogleCast:MediaPlaybackStarted";
+    protected static final String MEDIA_PLAYBACK_ENDED =
+            "GoogleCast:MediaPlaybackEnded";
+    protected static final String MEDIA_PROGRESS_UPDATED =
+            "GoogleCast:MediaProgressUpdated";
+
+    protected static final  String CHANNEL_MESSAGE_RECEIVED = "GoogleCast:Channel_message_received";
```
Comment (petrbela) [inline android/src/main/java/com/reactnative/googlecast/GoogleCastModule.java]: Please follow the CamelCase convention

File: android/src/main/java/com/reactnative/googlecast/GoogleCastModule.java [hunk 2]
```diff
@@ -14,288 +16,352 @@
...
+            getReactApplicationContext().runOnUiQueueThread(new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        mCastSession.setMessageReceivedCallbacks(namespace, new Cast.MessageReceivedCallback() {
+                            @Override
+                            public void onMessageReceived(CastDevice castDevice, String channelNameSpace, String message) {
+                                WritableMap map = Arguments.createMap();
+                                map.putString("getFriendlyName", castDevice.getFriendlyName());
+                                map.putString("nameSpace", channelNameSpace);
```
Comment (petrbela) [inline android/src/main/java/com/reactnative/googlecast/GoogleCastModule.java]: key should be called `channel`

File: android/src/main/java/com/reactnative/googlecast/GoogleCastModule.java [hunk 3]
```diff
@@ -14,288 +16,352 @@
...
+        if (mCastSession != null) {
+            getReactApplicationContext().runOnUiQueueThread(new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        mCastSession.setMessageReceivedCallbacks(namespace, new Cast.MessageReceivedCallback() {
+                            @Override
+                            public void onMessageReceived(CastDevice castDevice, String channelNameSpace, String message) {
+                                WritableMap map = Arguments.createMap();
+                                map.putString("getFriendlyName", castDevice.getFriendlyName());
```
Comment (petrbela) [inline android/src/main/java/com/reactnative/googlecast/GoogleCastModule.java]: This doesn't have iOS equivalent. Maybe just call it `deviceName` for now, it's being completely reworked in v4 anyway...

**Comment by ronit-kadwane** [comment]
Please check the changes 
I have changed all suggested naming convention and key name

File: android/src/main/java/com/reactnative/googlecast/GoogleCastModule.java [hunk 4]
```diff
@@ -59,7 +59,7 @@
     protected static final String MEDIA_PROGRESS_UPDATED =
             "GoogleCast:MediaProgressUpdated";
 
-    protected static final  String CHANNEL_MESSAGE_RECEIVED = "GoogleCast:Channel_message_received";
+    protected static final  String CHANNEL_MESSAGE_RECEIVED = "GoogleCast:ChannelMessageReceived";
```
Comment (ronit-kadwane) [inline android/src/main/java/com/reactnative/googlecast/GoogleCastModule.java]: CamelCase convention

File: android/src/main/java/com/reactnative/googlecast/GoogleCastModule.java [hunk 5]
```diff
@@ -206,8 +206,8 @@ public void run() {
                             @Override
                             public void onMessageReceived(CastDevice castDevice, String channelNameSpace, String message) {
                                 WritableMap map = Arguments.createMap();
-                                map.putString("getFriendlyName", castDevice.getFriendlyName());
-                                map.putString("nameSpace", channelNameSpace);
+                                map.putString("deviceName", castDevice.getFriendlyName());
```
Comment (ronit-kadwane) [inline android/src/main/java/com/reactnative/googlecast/GoogleCastModule.java]: Appropriate key

File: android/src/main/java/com/reactnative/googlecast/GoogleCastModule.java [hunk 6]
```diff
@@ -206,8 +206,8 @@ public void run() {
...
+                                map.putString("channel", channelNameSpace);
```
Comment (ronit-kadwane) [inline android/src/main/java/com/reactnative/googlecast/GoogleCastModule.java]: Appropriate key

**Comment by hillmanov** [comment]
@petrbela I just wanted to see if there is any kind of ETA for when this could get merged in? 

Thanks!

**Comment by petrbela** [comment]
Hey @hillmanov! I'm planning to work on RNGC this weekend, so I'll go through all of the PRs in one swoop.

**Pull request merged**